app/services/unpublish_service.rb

```diff
@@ -7,21 +7,17 @@ def withdraw(edition, public_explanation, user = nil)
       check_unpublishable(edition)
 
       previous_withdrawal = edition.withdrawn? && edition.status.details
-      withdrawal = if previous_withdrawal
-                     previous_withdrawal.dup.tap do |w|
-                       w.assign_attributes(public_explanation: public_explanation)
-                     end
-                   else
-                     Withdrawal.new(public_explanation: public_explanation,
-                                    published_status: edition.status,
-                                    withdrawn_at: Time.current)
-                   end
+
+      # Don't do anything if nothing has changed
+      return if previous_withdrawal && previous_withdrawal.public_explanation == public_explanation
```

Comment:
This plus comment suggests a private method to me ;-).

Reply:
Makes sense 👍